### ISSUE TITLE
freebsd fuse test fail

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#1734]
```
(borg-env) [vagrant@freebsd-10 /vagrant/borg/borg]$ sudo chmod 666 /dev/fuse
(borg-env) [vagrant@freebsd-10 /vagrant/borg/borg]$ py.test -k "fuse"
============================= test session starts ==============================
platform freebsd10 -- Python 3.5.2, pytest-3.0.3, py-1.4.31, pluggy-0.4.0
benchmark: 3.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=5.00us max_time=1.00s calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /vagrant/borg/borg, inifile: setup.cfg
plugins: cov-2.4.0, benchmark-3.0.0
collected 643 items 

borg/testsuite/archiver.py F.F.F.

=================================== FAILURES ===================================
__________________________ ArchiverTestCase.test_fuse __________________________

self = <borg.testsuite.archiver.ArchiverTestCase testMethod=test_fuse>

    @unittest.skipUnless(has_llfuse, 'llfuse not installed')
    def test_fuse(self):
        def has_noatime(some_file):
            atime_before = os.stat(some_file).st_atime_ns
            try:
                os.close(os.open(some_file, flags_noatime))
            except PermissionError:
                return False
            else:
                atime_after = os.stat(some_file).st_atime_ns
                noatime_used = flags_noatime != flags_normal
                return noatime_used and atime_before == atime_after
    
        self.cmd('init', self.repository_location)
        self.create_test_files()
        have_noatime = has_noatime('input/file1')
        self.cmd('create', self.repository_location + '::archive', 'input')
        self.cmd('create', self.repository_location + '::archive2', 'input')
        if has_lchflags:
            # remove the file we did not backup, so input and mount become equal
            os.remove(os.path.join('input', 'flagfile'))
        mountpoint = os.path.join(self.tmpdir, 'mountpoint')
        # mount the whole repository, archive contents shall show up in archivename subdirs of mountpoint:
        with self.fuse_mount(self.repository_location, mountpoint):
>           self.assert_dirs_equal(self.input_path, os.path.join(mountpoint, 'archive', 'input'))

borg/testsuite/archiver.py:1042: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
borg/testsuite/__init__.py:56: in assert_dirs_equal
    self._assert_dirs_equal_cmp(diff)
borg/testsuite/__init__.py:92: in _assert_dirs_equal_cmp
    self.assert_equal(d1, d2)
E   AssertionError: Lists differ: ['fifo1', 4480, 1001, 0, None, 2048, 1, 1476706254244000000, {}] != ['fifo1', 4480, 1001, 0, None, 0, 1, 1476706254244000000, {}]
E   
E   First differing element 5:
E   2048
E   0
E   
E   - ['fifo1', 4480, 1001, 0, None, 2048, 1, 1476706254244000000, {}]
E   ?                                - --
E   
E   + ['fifo1', 4480, 1001, 0, None, 0, 1, 1476706254244000000, {}]
_______________________ ArchiverTestCaseBinary.test_fuse _______________________

self = <borg.testsuite.archiver.ArchiverTestCaseBinary testMethod=test_fuse>

    @unittest.skipUnless(has_llfuse, 'llfuse not installed')
    def test_fuse(self):
        def has_noatime(some_file):
            atime_before = os.stat(some_file).st_atime_ns
            try:
                os.close(os.open(some_file, flags_noatime))
            except PermissionError:
                return False
            else:
                atime_after = os.stat(some_file).st_atime_ns
                noatime_used = flags_noatime != flags_normal
                return noatime_used and atime_before == atime_after
    
        self.cmd('init', self.repository_location)
        self.create_test_files()
        have_noatime = has_noatime('input/file1')
        self.cmd('create', self.repository_location + '::archive', 'input')
        self.cmd('create', self.repository_location + '::archive2', 'input')
        if has_lchflags:
            # remove the file we did not backup, so input and mount become equal
            os.remove(os.path.join('input', 'flagfile'))
        mountpoint = os.path.join(self.tmpdir, 'mountpoint')
        # mount the whole repository, archive contents shall show up in archivename subdirs of mountpoint:
        with self.fuse_mount(self.repository_location, mountpoint):
>           self.assert_dirs_equal(self.input_path, os.path.join(mountpoint, 'archive', 'input'))

borg/testsuite/archiver.py:1042: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
borg/testsuite/__init__.py:56: in assert_dirs_equal
    self._assert_dirs_equal_cmp(diff)
borg/testsuite/__init__.py:92: in _assert_dirs_equal_cmp
    self.assert_equal(d1, d2)
E   AssertionError: Lists differ: ['fifo1', 4480, 1001, 0, None, 2048, 1, 1476706259345000000, {}] != ['fifo1', 4480, 1001, 0, None, 0, 1, 1476706259345000000, {}]
E   
E   First differing element 5:
E   2048
E   0
E   
E   - ['fifo1', 4480, 1001, 0, None, 2048, 1, 1476706259345000000, {}]
E   ?                                - --
E   
E   + ['fifo1', 4480, 1001, 0, None, 0, 1, 1476706259345000000, {}]
_______________________ RemoteArchiverTestCase.test_fuse _______________________

self = <borg.testsuite.archiver.RemoteArchiverTestCase testMethod=test_fuse>

    @unittest.skipUnless(has_llfuse, 'llfuse not installed')
    def test_fuse(self):
        def has_noatime(some_file):
            atime_before = os.stat(some_file).st_atime_ns
            try:
                os.close(os.open(some_file, flags_noatime))
            except PermissionError:
                return False
            else:
                atime_after = os.stat(some_file).st_atime_ns
                noatime_used = flags_noatime != flags_normal
                return noatime_used and atime_before == atime_after
    
        self.cmd('init', self.repository_location)
        self.create_test_files()
        have_noatime = has_noatime('input/file1')
        self.cmd('create', self.repository_location + '::archive', 'input')
        self.cmd('create', self.repository_location + '::archive2', 'input')
        if has_lchflags:
            # remove the file we did not backup, so input and mount become equal
            os.remove(os.path.join('input', 'flagfile'))
        mountpoint = os.path.join(self.tmpdir, 'mountpoint')
        # mount the whole repository, archive contents shall show up in archivename subdirs of mountpoint:
        with self.fuse_mount(self.repository_location, mountpoint):
>           self.assert_dirs_equal(self.input_path, os.path.join(mountpoint, 'archive', 'input'))

borg/testsuite/archiver.py:1042: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
borg/testsuite/__init__.py:56: in assert_dirs_equal
    self._assert_dirs_equal_cmp(diff)
borg/testsuite/__init__.py:92: in _assert_dirs_equal_cmp
    self.assert_equal(d1, d2)
E   AssertionError: Lists differ: ['fifo1', 4480, 1001, 0, None, 2048, 1, 1476706268802000000, {}] != ['fifo1', 4480, 1001, 0, None, 0, 1, 1476706268802000000, {}]
E   
E   First differing element 5:
E   2048
E   0
E   
E   - ['fifo1', 4480, 1001, 0, None, 2048, 1, 1476706268802000000, {}]
E   ?                                - --
E   
E   + ['fifo1', 4480, 1001, 0, None, 0, 1, 1476706268802000000, {}]
```

**Comment by ThomasWaldmann** [comment]
Note: same may happen for "file1" also, so it is not specific to the FIFO.

**Comment by ThomasWaldmann** [comment]
This is due to flags not reflected by FUSE, from freebsd stat.h:
```
  283 #define UF_ARCHIVE      0x00000800      /* file needs to be archived */
```
This is the only flag that is always set for new files. FUSE does not reflect it (should it?). So we always have a mismatch on systems supporting (lch)flags.

**Comment by ThomasWaldmann** [comment]
I'll add ignore flags to assert_dirs_equal (defaulting to not ignoring stuff) that can be used to ignore owner/group/bsdflags/... differences if we expect them to be not equal due to circumstances that are not borg errors.

**Comment by ThomasWaldmann** [comment]
Fixed by PR #1756.
